javav2/example_code/iotsitewise/src/test/java/SitewiseTests.java

```diff
@@ -0,0 +1,260 @@
+public class SitewiseTests {
+
+    private static final String assetModelName = "MyAssetModel"+ UUID.randomUUID();
+    private static final String assetName = "MyAsset";
+
+    private static String assetId = "";
+    private static final String portalName = "MyPortal";
+    private static String contactEmail = "";
+    private static final String gatewayName = "myGateway"+ UUID.randomUUID();
+    private static final String myThing = "myThing"+ UUID.randomUUID();
+
+    private static String assetModelHello = "";
+
+    private static String assetModelId = "";
+
+    private static String iamRole = "";
+
+    private static final SitewiseActions sitewiseActions = new SitewiseActions();
+
+    private static Map<String, String> propertyIds;
+
+    private static String humPropId = "";
+
+    private static String tempPropId = "";
+
+    private static String portalId = "";
+
+    private static String gatewayId = "";
+    private static final String ROLES_STACK = "RoleSitewise";
+
+    @BeforeAll
+    public static void setUp() {
+        CloudFormationHelper.deployCloudFormationStack(ROLES_STACK);
+        Map<String, String> stackOutputs = CloudFormationHelper.getStackOutputs(ROLES_STACK);
+        iamRole = stackOutputs.get("SitewiseRoleArn");
+
+        Gson gson = new Gson();
+        String json = getSecretValues();
+        SecretValues values = gson.fromJson(json, SecretValues.class);
+        contactEmail = values.getContactEmail();
+        assetModelHello  = values.getAssetModelHello();
+    }
+
+    @Test
+    @Tag("IntegrationTest")
+    @Order(1)
+    public void testHelloService() {
+        assertDoesNotThrow(() -> HelloSitewise.fetchAssets(assetModelHello));
+        System.out.println(" Test 1 passed");
+    }
+
+    @Test
+    @Tag("IntegrationTest")
+    @Order(2)
+    public void testCreateAssetModel() {
+        assertDoesNotThrow(() -> {
+            CompletableFuture<CreateAssetModelResponse> future = sitewiseActions.createAssetModelAsync(assetModelName);
+            CreateAssetModelResponse response = future.join();
+
+            if (response == null || response.assetModelId() == null) {
+                throw new RuntimeException("Simulating failure: response or assetModelId is null");
+            }
+
+            // Simulate successful creation (won't reach this if above condition is met)
+            assetModelId = response.assetModelId();
+        });
+
+        // This won't be reached if the assert fails
+        System.out.println("Test 2 passed");
+    }
+
+    @Test
+    @Tag("IntegrationTest")
+    @Order(3)
+    public void testCreateAsset() throws InterruptedException {
+        Thread.sleep(30000);
+        assertDoesNotThrow(() -> {
+            CompletableFuture<CreateAssetResponse> future = sitewiseActions.createAssetAsync(assetName, assetModelId);
+            CreateAssetResponse response = future.join();
+            assetId = response.assetId();
```

Comment:
I think it would be better in all these test methods to test that the return value is what you expected rather than that the method under test just didn't throw an exception.

Reply:
LEts discuss